src/ui/src/core/templateMap.ts

```diff
@@ -163,8 +167,29 @@ function fallbackTemplate(type: string) {
 	};
 }
 
+function getMergedAbstractTemplate(type: string) {
+	const template = abstractTemplateMap[type];
+	if (!template) return;
+	const baseType = template.baseType;
+	return {
+		...templateMap[baseType],
+		writer: {
+			...templateMap[baseType].writer,
+			...abstractTemplateMap[type].writer,
+			fields: {
```

Comment:
Any better way to merge deep without causing trouble appreciated

Reply:
This pattern is usually call deep merge in javascript. I think there is simple way to do that with stdlib. You need to either build your own function deep merge function, either use library as lodash.

https://www.geeksforgeeks.org/how-to-deep-merge-two-objects-in-typescript/